MetaMask/metamask-extension
# Updating AvatarWithBadge to BadgeWrapper

## Explanation

Updating `AvatarWithBadge` to be called `BadgeWrapper` and updating component api according to insight report. Also introduces the first enums to the component library.


* Fixes #17706 
* [Insight report FigJam](https://www.figma.com/file/pQa2vI7r08pHiqsAPTWdQ8/Avatar%2C-Token%2C-Network%2C-Icon%2C-Badge-Audit?node-id=602%3A10&t=jOYT7v3ZGEeeyIfR-1)

<img width="460" alt="Screenshot 2023-02-21 at 9 50 20 PM" src="https://user-images.githubusercontent.com/8112138/220525047-674359b1-ea5a-4c3b-9810-a7cd42111ca4.png">


## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/8112138/220523096-b229db33-4238-4794-9a9b-98ae980e1cbc.mov

### After

https://user-images.githubusercontent.com/8112138/220523338-c2f9a6aa-f320-401a-a479-9863149338cb.mov

No instances of `AvatarWithBadge` in code base

<img width="320" alt="Screenshot 2023-02-21 at 9 28 56 PM" src="https://user-images.githubusercontent.com/8112138/220524816-6e1c9b8b-7766-48b7-b9d2-f2f63d2ce53d.png">

## Manual Testing Steps

- Go to the latest build of storybook in this PR
- Search for `BadgeWrapper` in the search bar in the side panel
- Check the storybook controls
- Check the stories render properly
- Check the documentation

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
